File: Problems-on-string-in-cpp/problem40.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////
// Write a program which accept string from user and count number of small characters.
//
// Input:
//
// "Marvellous"
//
//Output- 9
/////////////////////////////////////////////////////////////////////////////////////////////////


#include <iostream>
using namespace std;

int CountCapital(char *str)
{
    int count = 0;

    while (*str != '\0')
    {
        if (*str >= 'a' && *str <= 'z')
        {
            count++;
        }
        
        str++;
    }

    return count;
}

int main()
{
    char Arr[20];
    int iRet = 0;

    cout << "Enter string\n";
    cin.getline(Arr, 20);

    iRet = CountCapital(Arr);

    cout << "Number of small charecters in string are:\n"
         << iRet;

    return 0;
}

File: Problems-on-Bitwise/problem40.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// 
// Write a program which accept one number from user and count number of ON bits in it without using % or / operator
// 
// input : 11
// Output:- 3 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.lang.*;
import java.util.*;

class Bitwise
{
    int iValue=0;

    public Bitwise(int ino)                 // Parameterized constructor........
    {
        iValue=ino;
    }
    public int countonbits()
    {
        int iMask=0x00000001;
        int iResult=0;

        int iCnt=0;

        for(int i=1;i<=32;i++)
        {
            iResult=iValue & iMask;
            if(iResult==iMask)
            {
                iCnt++;
            }

            iMask=iMask<<1;
        }

        return iCnt;

    }

}



class program1_33
{
    public static void main(String args[])
    {
        Scanner sobj=new Scanner(System.in);
        System.out.println("Enter number");
        int iNo=sobj.nextInt();

        Bitwise bobj=new Bitwise(iNo);

        int iRet=bobj.countonbits();

        System.out.println("Number of on bits are"+iRet);



        

    }
}



